### PR TITLE
Implement build-harness support

### PR DESCRIPTION
**Pull request by goruha**
## what
* Use harness for docker tagging on travis

## why
* Facilitate long-term maintenance

## who
@cloudposse/engineering

### CODE REVIEW

File: Makefile.golang [hunk 1]
```diff
@@ -0,0 +1 @@
+GO:= $(shell which go)
```
Comment (osterman) [inline Makefile.golang]: Move to `build-harness/golang/Makefile`

File: Makefile.golang.build [hunk 2]
```diff
@@ -0,0 +1,48 @@
+GLIDE	:= $(shell which glide)
```
Comment (osterman) [inline Makefile.golang.build]: Move `Makefile.golang.build` to `build-harness/golang/Makefile.build`

File: Makefile.golang.style [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.PHONY: lint
```
Comment (osterman) [inline Makefile.golang.style]: Move to `build-harness/golang/Makefile.style`

File: Makefile [hunk 4]
```diff
@@ -1,61 +1,5 @@
-GO	:= $(shell which go)
-GLIDE	:= $(shell which glide)
-APP	?= github-authorized-keys
-INSTALL_DIR ?= /usr/local/sbin
-
 include Makefile.*
```
Comment (osterman) [inline Makefile]: add `-include build-harness/golang/Makefile*`

Make `build-harness` a submodule pinned to `master@HEAD` or add a target that runs `git clone`

File: Makefile.golang (hunk 1 above)
Comment (goruha) [inline Makefile.golang]: Done

File: Makefile.golang.build (hunk 2 above)
Comment (goruha) [inline Makefile.golang.build]: Done

File: Makefile.golang.style (hunk 3 above)
Comment (goruha) [inline Makefile.golang.style]: Done

File: Makefile (hunk 4 above)
Comment (goruha) [inline Makefile]: Done

File: Makefile.copyright [hunk 5]
```diff
@@ -1,21 +0,0 @@
-COPYRIGHT_CMD ?= docker run --rm --volume `pwd`:$(COPYRIGHT_OUTPUT_DIR) osterman/copyright-header:latest
-COPYRIGHT_LICENSE ?= ASL2
-COPYRIGHT_HOLDER ?= Cloud Posse, LLC <hello@cloudposse.com>
-COPYRIGHT_YEAR ?= 2016
-COPYRIGHT_SOFTWARE ?= Github Authorized Keys
```
Comment (osterman) [inline Makefile.copyright]: We still need to set this somewhere

File: Makefile.copyright [hunk 6]
```diff
@@ -1,21 +0,0 @@
...
-COPYRIGHT_SOFTWARE_DESCRIPTION ?= Use GitHub teams to manage system user accounts and authorized_keys
```
Comment (osterman) [inline Makefile.copyright]: and this

Comment (goruha) [inline Makefile.copyright]: Done

File: Makefile.copyright (hunk 5 above)
Comment (goruha) [inline Makefile.copyright]: Done

**Pull request merged**